server/app/lib/actions/fusor/content/manage_content.rb

```diff
@@ -26,30 +26,29 @@ def plan(deployment)
           fail _("Unable to locate puppet class settings in config/settings.plugins.d/fusor.yaml") unless SETTINGS[:fusor][:puppet_classes]
 
           sequence do
-            product_types = [:rhev, :cfme, :openstack]  # array of supported products
-            products_enabled = [deployment.deploy_rhev, deployment.deploy_cfme, deployment.deploy_openstack]
-
             content = SETTINGS[:fusor][:content]
-            products_content = [content[:rhev], content[:cloudforms], content[:openstack]]
             host_groups = SETTINGS[:fusor][:host_groups]
-            products_host_groups = [host_groups[:rhev], host_groups[:cloudforms], host_groups[:openstack]]
+            product_map = {rhev: [:rhevm, :rhevh],
+                           cfme: [:cloudforms],
+                           openstack: [:openstack]
+            }
 
             enable_smart_class_parameter_overrides
 
-            products_content_details = []
-            products_enabled.each_with_index do |product_enabled, i|
-              if product_enabled
-                products_content_details = (products_content_details << products_content[i]).flatten.uniq
+            product_content_details = product_map.flat_map do |deploy, product_types|
+              if deployment.deploy?(deploy)
+                product_types.flat_map { |type| content[type] }
               end
             end
+            product_content_details.compact!.uniq!
```

Comment:
compact! returns nil if no changes were made. Please separate this out into two lines to avoid an "undefined method `uniq!' for nil:NilClass" error.